crates/dojo-bindgen/src/plugins/recs/mod.rs

```diff
@@ -0,0 +1,618 @@
+                    }},
+                }}
+            );
+        }})(),
+",
+            path = model.type_path,
+            model = model.type_name(),
+            types = types.join(", "),
+            custom_types = custom_types.join(", ")
+        )
+    }
+
+    // Handles a model definition and its referenced tokens
+    // Will map all structs and enums to TS types
+    // Will format the models into a object
+    fn handle_model(&self, models: &[&DojoModel], handled_tokens: &mut Vec<Composite>) -> String {
+        let mut out = String::new();
+        out += TypescriptRecsPlugin::generated_header().as_str();
+        out += "import { defineComponent, Type as RecsType, World } from \"@dojoengine/recs\";\n";
+        out += "\n";
+        out += "export type ContractComponents = Awaited<ReturnType<typeof \
+                defineContractComponents>>;\n";
+
+        out += "\n\n";
+
+        let mut models_structs = Vec::new();
+        for model in models {
+            let tokens = &model.tokens;
+
+            let mut sorted_structs = tokens.structs.clone();
+            sorted_structs.sort_by(compare_tokens_by_type_name);
+
+            let mut sorted_enums = tokens.enums.clone();
+            sorted_enums.sort_by(compare_tokens_by_type_name);
+
+            for token in &sorted_enums {
+                handled_tokens.push(token.to_composite().unwrap().to_owned());
+            }
+            for token in &sorted_structs {
+                handled_tokens.push(token.to_composite().unwrap().to_owned());
+            }
+
+            for token in &sorted_enums {
+                if handled_tokens.iter().filter(|t| t.type_name() == token.type_name()).count() > 1
+                {
+                    continue;
+                }
+                out += TypescriptRecsPlugin::format_enum(token.to_composite().unwrap()).as_str();
+            }
+
+            for token in &sorted_structs {
+                if handled_tokens.iter().filter(|t| t.type_name() == token.type_name()).count() > 1
+                {
+                    continue;
+                }
+
+                // first index is our model struct
+                if token.type_name() == naming::get_name_from_tag(&model.tag) {
+                    models_structs.push((
+                        naming::get_namespace_from_tag(&model.tag),
+                        token.to_composite().unwrap().clone(),
+                    ));
+                }
+
+                out += TypescriptRecsPlugin::format_struct(token.to_composite().unwrap()).as_str();
+            }
+
+            out += "\n";
+        }
+
+        out += "
+export function defineContractComponents(world: World) {
+    return {
+";
+
+        for (namespace, model) in models_structs {
+            out += TypescriptRecsPlugin::format_model(&namespace, &model).as_str();
+        }
+
+        out += "    };
+}\n";
+
+        out
+    }
+
+    // Formats a system into a C# method used by the contract class
```

Comment:
```suggestion
    // Formats a system into a Typescript method used by the contract class
```
In a next pass, could be helpful having an adjustment of comments to the very specific use-case here to contextualize more the reader.